nics-tw/PETsARD
# fix: Smartnoise could be abled synthesizing based on parameter and Metadata

- See #548 
- 修改方向：
    - 檢查是不是 Smartnoise 的 sample 用的是參數、而沒有用 `self.`
    - 檢查後發現不是這問題，是 Smartnoise 的工廠方法沒有帶入 Metadata

![image](https://github.com/nics-tw/PETsARD/assets/8596186/b23e75e0-4712-41d6-8fb5-3c08fe35c70d)

---

- fix
    - PETsARD/operator.py
        - assign sample_dict for SynthesizerOperator sample() - 88b77455f355fc6d1fc4396db94d345d9a086b0a 
    - PETsARD/synthesizer/synthesizer.py
        - simplify Synthesizer fit() parameter - c867ccb900a87b625ffbb6e68271dce6f3227f1b 
    - PETsARD/synthesizer/smartnoise.py
        - make Smartnoise load metadata for SynthesizerBase - acc7e4c670416a986198734801e19bec11f482ec 

- test
    - /demo/dev/Issue548.ipynb
        - create Issue548.ipynb - ecfb20592cb906c31bf22b9042b0bfbe9f32d2bf 
        - update success fixation - ff3d3b4606d3f5551990b001839ce2d1de1e6938 
        - delete Issue548.ipynb - f391ed2cd0cfbdbed343334e405f872fcfe9c8aa 

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207341981927501
